Lombiq.HelpfulLibraries/Libraries/Shapes/ShapeExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using OrchardCore.DisplayManagement;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Lombiq.HelpfulLibraries.Libraries.Shapes;
+
+public static class ShapeExtensions
+{
+    /// <summary>
+    /// Creates a new instance of <typeparamref name="T"/> and sets its properties from the <paramref name="shape"/>'s
+    /// <see cref="IShape.Properties"/> with the same keys.
+    /// </summary>
+    public static T As<T>(this IShape shape)
+        where T : new()
+    {
+        var result = new T();
+
+        var shapeProperties = shape.GetType().GetProperties().ToDictionary(property => property.Name);
+        foreach (var property in typeof(T).GetProperties())
+        {
+            if ((shape.Properties.GetMaybe(property.Name) ?? shapeProperties.GetMaybe(property.Name)?.GetValue(shape)) is { } value)
+            {
+                property.SetValue(result, value);

```

Comment:
Shouldn't you throw an `ArgumentException` if the property types don't match?

Reply:
I just wanted to automate away the 
```
var property1 = @Model.Propert1 as string;
var property2 = @Model.Property2 as bool? == true;
...
```
kind of stuff from the cshtml files so I didn't consider additional constraints. Why do you think it's necessary?

Reply:
So it'll blow up quickly and with a helpful error message in case a developer casts to the wrong type.

Reply:
Ah, I assumed it already does that. After actually checking (below) looks like it only gives you the received and expected types, but not the property name! That's pretty dumb.
![image](https://user-images.githubusercontent.com/4007293/160213037-9652bf33-1b07-48db-81f8-54f581a14cfc.png)

Reply:
I agree. Same with duplicate Dictionary keys.

I guess it's some security measure - you never know what people pass around 😉

Reply:
How do you start that C# repl in PS, btw?

Reply:
It's called [dotnet-script](https://github.com/filipw/dotnet-script). You can install it with `dotnet tool install -g dotnet-script` and then execute with `dotnet script`. It can also run C# script files (kinda like the new [top-level statements](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/program-structure/top-level-statements), but without even compilation) and import libraries from DLL or NuGet directly. There is no tab completion or intellisense though.
I use it or pwsh depending on the mood :)

Reply:
Thanks! I'll take for a ride :-D